src/r1cs_nark_pcd/data_structures.rs

```diff
@@ -0,0 +1,80 @@
+use ark_accumulation::r1cs_nark_as;
+use ark_accumulation::r1cs_nark_as::{r1cs_nark, AccumulatorInstance, AccumulatorWitness};
+use ark_ec::{AffineCurve, CurveCycle};
+
+pub(crate) type MainAffine<E> = <E as CurveCycle>::E1;
+pub(crate) type HelpAffine<E> = <E as CurveCycle>::E2;
+
+pub(crate) type MainField<E> = <<E as CurveCycle>::E2 as AffineCurve>::BaseField;
```

Comment:
```suggestion
pub(crate) type MainField<E> = <<E as CurveCycle>::E1 as AffineCurve>::ScalarField;
```

Reply:
For some reason, applying this change causes the library to not compile. An example of an error is: `the trait ark_ff::PrimeField is not implemented for <<E as CurveCycle>::E2 as AffineCurve>::BaseField`. It isn't clear to me why it isn't detecting it as a `PrimeField`. Maybe it is just substituting the `BaseField` in without doing anything else?